specs/directory.md

```diff
@@ -0,0 +1,375 @@
+
+### System Level
+
+Default directory paths for various operating systems at system level are specified as below.
+
+| Directory | Unix           | Windows                       | Darwin                         |
+| --------- | -------------- | ----------------------------- | ------------------------------ |
+| `BIN`     | `/usr/bin`     | `%ProgramFiles%/notation/bin` | `/usr/local/bin`               |
+| `LIBEXEC` | `/usr/libexec` | `%ProgramFiles%`              | `/usr/local/lib`               |
+| `CONFIG`  | `/etc`         | `%ProgramData%`               | `/Library/Application Support` |
+
+`CACHE` is omitted since it is user specific.
+
+### User Level
+
+Default directory paths for various operating systems at user level are specified as below.
+
+| Directory | Unix               | Windows          | Darwin                          |
+| --------- | ------------------ | ---------------- | ------------------------------- |
+| `LIBEXEC` | `$XDG_CONFIG_HOME` | `%AppData%`      | `~/Library/Application Support` |
+| `CONFIG`  | `$XDG_CONFIG_HOME` | `%AppData%`      | `~/Library/Application Support` |
+| `CACHE`   | `$XDG_CACHE_HOME`  | `%LocalAppData%` | `~/Library/Caches`              |
+
+There is no default `BIN` path at user level since the `notation` binary can be put anywhere as long as it in the `PATH` environment variable. Common directories on Unix/Darwin are `~/bin` and `~/.local/bin` where manual `PATH` update by users may be required.
+
+## Structure
+
+The overall directory structure for `notation` is summarized as follows.
+
+```
+{BIN}
+└── notation
+{CACHE}
+└── notation
+    └── signature
+        └── {manifest-digest-algorithm}
+            └── {manifest-digest}
+                └── {signature-digest-algorithm}
+                    └── {signature-digest}.sig
+{CONFIG}
+└── notation
+    ├── config.json
+    ├── signing
+    │   ├── certificates
+    │   │   └── {key-name}.crt
+    │   ├── keys
+    │   │   └── {key-name}.pem
+    │   └── keys.json
+    └── trust
+        ├── policy.json
+        └── store
+            └── {trust-store-type}
+                └── {named-store}
+                    └── {cert-file}
+{LIBEXEC}
+└── notation
+    └── plugins
+        └── {plugin-name}
+            └── notation-{plugin-name}
+```
+
+### Notation Binary
+
+The path for the `notation` binary is
+
+```
+{BIN}/notation
+```
+
+### Plugin
+
+[Plugins][Plugin] are binaries not meant to be executed directly by users' shell or scripts. The path of a plugin follows the pattern below.
+
+```
+{LIBEXEC}/notation/plugins/{plugin-name}/notation-{plugin-name}

```

Comment:
Should we note that `windows` binaries must end with `.exe`? The extension is not part of `{plugin-name}`, so it can cause confusion for Windows developers reading this spec.

Reply:
Good catch. I'll also update the Windows example.